packages/react-component-library/src/components/Breadcrumbs/BreadcrumbsItem.tsx

```diff
@@ -16,28 +16,51 @@ export interface BreadcrumbsItemProps {
    * @private
    */
   isLast?: boolean
+}
+
+export interface BreadcrumbsItemWithLinkProps extends BreadcrumbsItemBaseProps {
   /**
    * Link component to use for the item (custom implementation welcome).
    */
   link: React.ReactElement<LinkTypes>
+  href?: never
 }
 
-function getText(isLast: boolean, link: React.ReactElement<LinkTypes>) {
+export interface BreadcrumbsItemWithHrefProps extends BreadcrumbsItemBaseProps {
+  /**
+   * Href attribute to apply to HTML anchor
+   */
+  href: string
+  link?: never
+}
```

Comment:
Would be good to make it clear in the doc comments that `href` isn't needed if `link` is used (and vice versa) so it's clearer in Storybook:

![image](https://user-images.githubusercontent.com/66470099/149158428-ef4f9fff-4788-4b5c-9f6b-1d9a778b4d78.png)